### PR TITLE
Delete the deprecated data_type_id_t enum

### PR DESCRIPTION
**Pull request by ChuckHastings**
Prep work for temporal graphs.  We renamed `data_type_id_t` to `cugraph_data_id_t` about 8 months ago, this finishes the transition.  We also split the types out into their own file in the C API.

Marked as breaking since we are deleting a deprecated type from the C API.

### CODE REVIEW

File: cpp/include/cugraph_c/types.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+extern "C" {
+#endif
+
+typedef enum bool_ { FALSE = 0, TRUE = 1 } bool_t;
+
+typedef int8_t byte_t;
+
+typedef enum data_type_id_ {
+  INT32 = 0,
+  INT64,
```
Comment (seunghwak) [inline cpp/include/cugraph_c/types.h:31]: No need to add INT8, INT16, UINT8, UINT16, UINT32, UINT64?

These may become necessary in some algorithms in the future.

Comment (seunghwak) [reply cpp/include/cugraph_c/types.h:31]: If we add these types but don't explicitly instantiate for these types, the cost is pretty minimal, right?

**Review by seunghwak — APPROVED** [review]
Besides adding more integer types to be future proof, LGTM.

File: cpp/include/cugraph_c/types.h (hunk 1 above)
Comment (ChuckHastings) [reply cpp/include/cugraph_c/types.h:33]: We can keep the cost minimal.  I have added them in the next push.

**Review by rlratzel — APPROVED** [review]
cython/python changes LGTM, CI should tell us if anything was missed.

**Pull request merged**